TrustSDK/Classes/Client/Types/TrustError.swift

```diff
@@ -0,0 +1,78 @@
+//  Copyright © 2018 Trust.
+//
+//	This file is part of TrustSDK. The full TrustSDK copyright notice, including
+//	terms governing use, modification, and redistribution, is contained in the
+//	file LICENSE at the root of the source code distribution tree.
+	
+
+import Foundation
+
+public enum TrustSDKError: Swift.Error {
+    case notInitialized
+    case coinNotSupported
+    case invalidResponse
+    case rejectedByUSer
+    case signError(message: String)
+    case unknown
+    case invalidWallet
+    
+    init?(from name: String, value: String? = nil) {
+        switch name {
+        case "notInitialized":
+            self = .notInitialized
+        case "coinNotSupported":
+            self = .coinNotSupported
+        case "invalidResponse":
+            self = .invalidResponse
+        case "rejectedByUSer":
+            self = .rejectedByUSer
+        case "signError":
+            self = .signError(message: value ?? "")
+        case "unknown":
+            self = .unknown
+        case "invalidWallet":
+            self = .invalidWallet
+        default:
+            return nil
+        }
+    }
+    
+    init?(components: URLComponents) {
+        guard let name = components.queryItem(for: "error")?.value else { return nil }
+        self.init(from: name, value: components.queryItem(for: "message")?.value)
+    }
+        
+    public var description: String {
+        switch self {
+        case .notInitialized:
+            return "notInitialized"
```

Comment:
can you add strings to enums to avoid duplication?

Reply:
Done